In [2]:
import pandas as pd

proverbs = pd.read_csv("../csv/bonus_assignment.csv")
proverbs = proverbs.drop(columns='bonus_clustering')

proverbs

text
0                                   Γέλια σαν κομπολόγια
1                            Καρδίαν καθαράν θέλ' ο Θεός
2                                 Ου Θεός κι ου γείτονας
3                Θέλει να κρυφθή πίσω από το δάχτυλό του
4          Όλοι κλαίν' τα χάλια τ'ς κι ο μυλωνάς τη δέσι
...                                                  ...
11495                          Παστρζικό τσανα τσ' έναι!
11496        Κάλλιο γεναίκα κάμισσα, πέρζι πολυπροικούσα
11497  Όγοιος κάθεται στη στερζά τσαί θάλασσα γερεύει...
11498                          Πήε στ' δαιμόνου τ' μάννα
11499                   Θα σε κάνου να κατουρζήσεις αίμα

[11500 rows x 1 columns]

In [3]:
"""
    Creator of grecka module : ThemisB
    github-repo : https://github.com/ThemisB/grecka.git
"""
import sys
import os


current_dir = os.path.dirname(os.path.abspath(__file__)) if "__file__" in locals() else os.getcwd()
sys.path.append(os.path.join(current_dir, ".."))

# Import greeklish word transformer
from grecka.lib.grecka import Grecka


# Transform proverbs to greeklish
proverbs['greeklish_version'] = proverbs['text'].apply(Grecka.toGreeklish)

In [4]:
concat_df = pd.DataFrame(columns=['text'])
# Concatenate the original 'text' and 'greeklish_version' columns
concat_df['text'] = pd.concat([proverbs['text'], proverbs['greeklish_version']], axis=0, ignore_index=True)

In [5]:
# Assign the orig languages
concat_df['orig_lang'] = 'Greek'
concat_df.loc[11500:, 'orig_lang'] = 'Greeklish'

concat_df

text  orig_lang
0                                   Γέλια σαν κομπολόγια      Greek
1                            Καρδίαν καθαράν θέλ' ο Θεός      Greek
2                                 Ου Θεός κι ου γείτονας      Greek
3                Θέλει να κρυφθή πίσω από το δάχτυλό του      Greek
4          Όλοι κλαίν' τα χάλια τ'ς κι ο μυλωνάς τη δέσι      Greek
...                                                  ...        ...
22995                          Pastrziko tsana ts' enai!  Greeklish
22996        Kallio genaika kamissa, perzi polyproikousa  Greeklish
22997  Ogoios kathetai sti sterza tsai thalassa gerey...  Greeklish
22998                          Pie st' daimonou t' manna  Greeklish
22999                  Tha se kanou na katourziseis aima  Greeklish

[23000 rows x 2 columns]

In [6]:
kaggle_set = pd.read_csv("../csv/kaggle.csv")
kaggle_set


import pandas as pd

# Assuming 'kaggle_set' is your DataFrame
def categorize_language(language):
    if language == 'Greek':
        return 'Greek'
    elif language == 'English':
        return 'English'
    else:
        return 'Other'

kaggle_set['new_lang'] = kaggle_set['Language'].apply(categorize_language)
print(kaggle_set.new_lang.value_counts())

new_lang
Other      8587
English    1385
Greek       365
Name: count, dtype: int64


In [14]:
result_df = pd.DataFrame(columns=['text', 'orig_lang'])
# Concatenate the original 'text' and 'greeklish_version' columns
result_df['text'] = pd.concat([concat_df['text'], kaggle_set['Text']], axis=0, ignore_index=True)
result_df['orig_lang'] = pd.concat(
    [concat_df['orig_lang'], kaggle_set['new_lang']], axis=0, ignore_index=True)

result_df

text orig_lang
0                                   Γέλια σαν κομπολόγια     Greek
1                            Καρδίαν καθαράν θέλ' ο Θεός     Greek
2                                 Ου Θεός κι ου γείτονας     Greek
3                Θέλει να κρυφθή πίσω από το δάχτυλό του     Greek
4          Όλοι κλαίν' τα χάλια τ'ς κι ο μυλωνάς τη δέσι     Greek
...                                                  ...       ...
33332  ನಿಮ್ಮ ತಪ್ಪು ಏನು ಬಂದಿದೆಯೆಂದರೆ ಆ ದಿನದಿಂದ ನಿಮಗೆ ಒ...     Other
33333  ನಾರ್ಸಿಸಾ ತಾನು ಮೊದಲಿಗೆ ಹೆಣಗಾಡುತ್ತಿದ್ದ ಮಾರ್ಗಗಳನ್...     Other
33334  ಹೇಗೆ ' ನಾರ್ಸಿಸಿಸಮ್ ಈಗ ಮರಿಯನ್ ಅವರಿಗೆ ಸಂಭವಿಸಿದ ಎ...     Other
33335  ಅವಳು ಈಗ ಹೆಚ್ಚು ಚಿನ್ನದ ಬ್ರೆಡ್ ಬಯಸುವುದಿಲ್ಲ ಎಂದು ...     Other
33336  ಟೆರ್ರಿ ನೀವು ನಿಜವಾಗಿಯೂ ಆ ದೇವದೂತನಂತೆ ಸ್ವಲ್ಪ ಕಾಣು...     Other

[33337 rows x 2 columns]

In [70]:
result_df.to_csv("../csv/gold.csv", index=False, encoding='utf-8')


In [71]:
import pandas as pd

english_proverbs = pd.read_csv("../csv/English_phrases_and_sayings.csv")
print(english_proverbs.origin.value_counts())
print(english_proverbs.columns)

origin
English    2298
Name: count, dtype: int64
Index(['Unnamed: 0', 'text', 'explanation', 'origin'], dtype='object')


So, there are 2298 english proverbs in this dataset

In [72]:
final_df = pd.DataFrame(columns=["text", "orig_lang"])


# Concatenate the original 'text' and 'greeklish_version' columns
final_df['text'] = pd.concat([result_df['text'], english_proverbs['text']], axis=0, ignore_index=True)
final_df['orig_lang'] = pd.concat(
    [result_df['orig_lang'], english_proverbs['origin']], axis=0, ignore_index=True)

In [73]:
# Display the dataframe
final_df

text orig_lang
0                               Γέλια σαν κομπολόγια     Greek
1                        Καρδίαν καθαράν θέλ' ο Θεός     Greek
2                             Ου Θεός κι ου γείτονας     Greek
3            Θέλει να κρυφθή πίσω από το δάχτυλό του     Greek
4      Όλοι κλαίν' τα χάλια τ'ς κι ο μυλωνάς τη δέσι     Greek
...                                              ...       ...
35630                        Your money or your life   English
35631                               Your name is mud   English
35632                    You've never had it so good   English
35633                                 Zero tolerance   English
35634                                        Zig-zag   English

[35635 rows x 2 columns]

In [74]:
final_df.orig_lang.value_counts()

orig_lang
Greek        11865
Greeklish    11500
Other         8587
English       3683
Name: count, dtype: int64

    We can observe that the final dataset is highly unbalance on the number of labels for each class.

    This can result to unwanted biases and extra operations to reduce bias.
    
    So, given that we have the freedom to augment our dataset we will therefore select an equal amount of class labels.

In [75]:
# Store the minimum value of the value_counts() result
min_count = final_df['orig_lang'].value_counts().min()

# Sample rows from each category to match the count of the category
# with the lowest count
balanced_df = final_df.groupby('orig_lang').apply(lambda x: x.sample(min_count)).reset_index(drop=True)

In [76]:
balanced_df.orig_lang.value_counts()

orig_lang
English      3683
Greek        3683
Greeklish    3683
Other        3683
Name: count, dtype: int64

Drop a portion of 'other' labeled rows, as it not of particular interest for the coming objective.

In [77]:
import pandas as pd

# Assuming 'balanced_df' is your DataFrame
percentage_to_drop = 0.5  # adjust the percentage as needed

# Get the indices of rows labeled as 'Other'
other_indices = balanced_df[balanced_df['orig_lang'] == 'Other'].index

# Convert the Index object to a pandas Series and then randomly select a percentage of indices to drop
indices_to_drop = pd.Series(other_indices).sample(frac=percentage_to_drop)

# Drop the selected rows
balanced_df_dropped = balanced_df.drop(indices_to_drop)

# Verify the new distribution
print(balanced_df_dropped['orig_lang'].value_counts())


orig_lang
English      3683
Greek        3683
Greeklish    3683
Other        1841
Name: count, dtype: int64


In [78]:
balanced_df_dropped.to_csv("../csv/gold.csv", index=False, encoding='utf-8')
